### PR TITLE
Create a unit-test job in CI.yaml

### PR DESCRIPTION
**Pull request by KubaKrzych**
Added a job that tests the code.

### CODE REVIEW

**Comment by KubaKrzych** [comment]
@matined @TheRealSeber 
The CI pipeline runs just fine. I've decided to add caching the dependencies, but I've not found it very useful, as the process of collecting them does not take that much time, but installing them is what takes the longest.

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -32,3 +32,51 @@ jobs:
...
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-${{ hashFiles('**/requirements.txt' )}}
+
+      - name: Prepare and Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt pytest httpx
```
Comment (matined) [inline .github/workflows/CI.yaml:73]: add pytest and httpx to requirements.txt

File: .github/workflows/CI.yaml [hunk 2]
```diff
@@ -32,3 +32,51 @@ jobs:
...
+
+      - name: Unit tests
+        run: |
+          pytest
```
Comment (matined) [inline .github/workflows/CI.yaml:77]: can we add `-vv` flags?

File: .github/workflows/CI.yaml [hunk 3]
```diff
@@ -32,3 +32,51 @@ jobs:
...
+        env: 
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_USER: postgres
+          POSTGRES_DB: postgres
+
```
Comment (matined) [inline .github/workflows/CI.yaml:82]: Additionally to checking if tests pass we want to check and report code coverage. We can use `pytest-cov` for calculating and generating a html report. Let's also check if github workflows offers a possibility to publish the report.

**Review by matined — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yaml (hunk 3 above)
Comment (KubaKrzych) [reply .github/workflows/CI.yaml]: I've found an interesting GitHub action that writes a comment whenever a pull_request is made. You can see an example below.

**Comment by KubaKrzych** [comment]
@matined Could I ask you for the review of the changes I've made? I've implemented everything you had listed, but I'm not sure whether I did so in a way you've wanted.

File: .github/workflows/CI.yaml (hunk 3 above)
Comment (matined) [reply .github/workflows/CI.yaml]: Yeah, that's what we want. However, we still cannot check which lines are covered, right?

**Review by matined — COMMENTED** [review]
Great! Can we check if we can somehow see which lines are covered?

**Comment by KubaKrzych** [comment]
> Great! Can we check if we can somehow see which lines are covered?

I see what you were referring to. I've added the report of uncovered lines in tests.

**Pull request merged**